Can't run CI on Fedora 42 due to missing headers of gobject dependencies

### Qubes OS release

Qubes OS 4.3

### Brief summary

It seems that Fedora-42 CI is missing a dependency, most likely [cairo-gobject-devel/](https://packages.fedoraproject.org/pkgs/cairo/cairo-gobject-devel/).

### Steps to reproduce

1. Trigger Gitlab CI multiple times by subimitting changes to a repo, such as in qubes-core-qrexec


### Expected behavior

No build failures, as in Fedora 40, ramen-vm: https://gitlab.com/QubesOS/qubes-core-qrexec/-/jobs/13276613570

### Actual behavior

Fedora 42, gitlab-ci-fraise: https://gitlab.com/QubesOS/qubes-core-qrexec/-/jobs/13275953718

```
$ pip3 install --user --quiet -r ci/requirements.txt
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [130 lines of output]
      + meson setup /tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f /tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f/.mesonpy-ifsst622 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md -Dtests=false -Dwheel=true --wrap-mode=nofallback --native-file=/tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f/.mesonpy-ifsst622/meson-python-native-file.ini
      The Meson build system
      Version: 1.10.1
      Source dir: /tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f
      Build dir: /tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f/.mesonpy-ifsst622
      Build type: native build
      Project name: pygobject
      Project version: 3.48.2
      C compiler for the host machine: cc (gcc 15.2.1 "cc (GCC) 15.2.1 20260123 (Red Hat 15.2.1-7)")
      C linker for the host machine: cc ld.bfd 2.44-12
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Program python3 found: YES (/usr/bin/python3)
      Found pkg-config: YES (/usr/sbin/pkg-config) 2.3.0
      Run-time dependency python found: YES 3.13
      Found pkg-config: YES (/usr/sbin/pkg-config) 2.3.0
      Run-time dependency gobject-introspection-1.0 found: YES 1.84.0
      Run-time dependency glib-2.0 found: YES 2.84.4
      Run-time dependency gobject-2.0 found: YES 2.84.4
      Run-time dependency gio-2.0 found: YES 2.84.4
      Run-time dependency gmodule-2.0 found: YES 2.84.4
      Run-time dependency libffi found: YES 3.4.6
      Run-time dependency cairo found: YES 1.18.2
      Did not find CMake 'cmake'
      Found CMake: NO
      Run-time dependency cairo-gobject found: NO (tried pkgconfig and cmake)
      Message: Found pycairo with target Python: /tmp/pip-build-env-v8bd4xv6/overlay/lib64/python3.13/site-packages/cairo/include
      Compiler for C supports arguments -Wall: YES
      Compiler for C supports arguments -Warray-bounds: YES
      Compiler for C supports arguments -Wcast-align: YES
      Compiler for C supports arguments -Wduplicated-branches: YES
      Compiler for C supports arguments -Wextra: YES
      Compiler for C supports arguments -Wformat=2: YES
      Compiler for C supports arguments -Wformat-nonliteral: YES
      Compiler for C supports arguments -Wformat-security: YES
      Compiler for C supports arguments -Wimplicit-function-declaration: YES
      Compiler for C supports arguments -Winit-self: YES
      Compiler for C supports arguments -Wjump-misses-init: YES
      Compiler for C supports arguments -Wlogical-op: YES
      Compiler for C supports arguments -Wmissing-declarations: YES
      Compiler for C supports arguments -Wmissing-format-attribute: YES
      Compiler for C supports arguments -Wmissing-include-dirs: YES
      Compiler for C supports arguments -Wmissing-noreturn: YES
      Compiler for C supports arguments -Wmissing-prototypes: YES
      Compiler for C supports arguments -Wnested-externs: YES
      Compiler for C supports arguments -Wnull-dereference: YES
      Compiler for C supports arguments -Wold-style-definition: YES
      Compiler for C supports arguments -Wpacked: YES
      Compiler for C supports arguments -Wpointer-arith: YES
      Compiler for C supports arguments -Wrestrict: YES
      Compiler for C supports arguments -Wreturn-type: YES
      Compiler for C supports arguments -Wshadow: YES
      Compiler for C supports arguments -Wsign-compare: YES
      Compiler for C supports arguments -Wstrict-aliasing: YES
      Compiler for C supports arguments -Wstrict-prototypes: YES
      Compiler for C supports arguments -Wswitch-default: YES
      Compiler for C supports arguments -Wundef: YES
      Compiler for C supports arguments -Wunused-but-set-variable: YES
      Compiler for C supports arguments -Wwrite-strings: YES
      Compiler for C supports arguments -Wno-incompatible-pointer-types-discards-qualifiers: NO
      Compiler for C supports arguments -Wno-missing-field-initializers: YES
      Compiler for C supports arguments -Wno-unused-parameter: YES
      Compiler for C supports arguments -Wno-discarded-qualifiers: YES
      Compiler for C supports arguments -Wno-sign-conversion: YES
      Compiler for C supports arguments -Wno-cast-function-type: YES
      Compiler for C supports arguments -Wno-int-conversion: YES
      Compiler for C supports arguments -fno-strict-aliasing: YES
      Compiler for C supports arguments -fvisibility=hidden: YES
      Configuring config.h using configuration
      Build targets in project: 2
      
      pygobject 3.48.2
      
        User defined options
          Native files: /tmp/pip-install-r4k2w0w3/pygobject_aad3d334fcc64321bd513b462ee7411f/.mesonpy-ifsst622/meson-python-native-file.ini
          b_ndebug    : if-release
          b_vscrt     : md
          buildtype   : release
          tests       : false
          wheel       : true
          wrap_mode   : nofallback
      
      Found ninja-1.13.0.git.kitware.jobserver-pipe-1 at /tmp/pip-build-env-v8bd4xv6/normal/bin/ninja
      + /tmp/pip-build-env-v8bd4xv6/normal/bin/ninja
      [1/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygboxed.c.o
      [2/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygflags.c.o
      [3/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygenum.c.o
      [4/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pyginterface.c.o
      [5/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygparamspec.c.o
      [6/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygpointer.c.o
      [7/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygoptioncontext.c.o
      [8/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygoptiongroup.c.o
      [9/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygspawn.c.o
      [10/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygobject-object.c.o
      [11/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-repository.c.o
      [12/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign.c.o
      [13/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/gimodule.c.o
      [14/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-struct.c.o
      [15/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-source.c.o
      [16/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-resulttuple.c.o
      [17/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-info.c.o
      [18/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-argument.c.o
      [19/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-boxed.c.o
      [20/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-ccallback.c.o
      [21/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-closure.c.o
      [22/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-type.c.o
      [23/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-util.c.o
      [24/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-property.c.o
      [25/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-signal-closure.c.o
      [26/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-invoke.c.o
      [27/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-marshal-cleanup.c.o
      [28/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-cache.c.o
      [29/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-list.c.o
      [30/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-basictype.c.o
      [31/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-array.c.o
      [32/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-error.c.o
      [33/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-object.c.o
      [34/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-fundamental.c.o
      [35/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-enum-marshal.c.o
      [36/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-value.c.o
      [37/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-hashtable.c.o
      [38/41] Compiling C object gi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign-cairo.c.o
      FAILED: [code=1] gi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign-cairo.c.o
      cc -Igi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p -Igi -I../gi -I/tmp/pip-build-env-v8bd4xv6/overlay/lib64/python3.13/site-packages/cairo/include -I/usr/include/python3.13 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/gobject-introspection-1.0 -I/usr/include/cairo -I/usr/include/libxml2 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/pixman-1 -fvisibility=hidden -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fPIC -DWITH_GZFILEOP -pthread -DPY_SSIZE_T_CLEAN -Wall -Warray-bounds -Wcast-align -Wduplicated-branches -Wextra -Wformat=2 -Wformat-nonliteral -Wformat-security -Wimplicit-function-declaration -Winit-self -Wjump-misses-init -Wlogical-op -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wpacked -Wpointer-arith -Wrestrict -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wundef -Wunused-but-set-variable -Wwrite-strings -Wno-missing-field-initializers -Wno-unused-parameter -Wno-discarded-qualifiers -Wno-sign-conversion -Wno-cast-function-type -Wno-int-conversion -fno-strict-aliasing -fvisibility=hidden -MD -MQ gi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign-cairo.c.o -MF gi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign-cairo.c.o.d -o gi/_gi_cairo.cpython-313-x86_64-linux-gnu.so.p/pygi-foreign-cairo.c.o -c ../gi/pygi-foreign-cairo.c
      ../gi/pygi-foreign-cairo.c:28:10: fatal error: cairo-gobject.h: No such file or directory
         28 | #include <cairo-gobject.h>
            |          ^~~~~~~~~~~~~~~~~
      compilation terminated.
      [39/41] Compiling C object gi/_gi.cpython-313-x86_64-linux-gnu.so.p/pygi-struct-marshal.c.o
      ninja: build stopped: subcommand failed.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
× Encountered error while generating package metadata.
╰─> See above for output.
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

### Additional information

_No response_

--- a/runner/docker/qubesos-ci/Dockerfile
+++ b/runner/docker/qubesos-ci/Dockerfile
@@ -12,6 +12,7 @@ RUN dnf -y install \
     e2fsprogs \
     gcc-c++ \
     git \
+    gobject-introspection-devel \
     gpg \
     grub2-tools \
     hunspell \

--- a/runner/kvm/packages_fedora.list
+++ b/runner/kvm/packages_fedora.list
@@ -12,6 +12,7 @@ gcc-c++
 git
 git-lfs
 gitlab-runner
+gobject-introspection-devel
 gpg
 grub2-tools
 gtk-layer-shell

